volumes.c

```diff
@@ -240,15 +240,21 @@ int pv_volume_mount(struct pv_volume *v)
 					PV_VOLMOUNT_DM_EXTRA_ARGS);
 
 			if (verity_options == NULL)
-				asprintf(&command, "%s mount %s %s %s",
-					 script, partname, path, name);
+				ret = asprintf(&command, "%s mount %s %s %s",
+					       script, partname, path, name);
 			else
-				asprintf(&command, "%s mount %s %s %s %s",
-					 script, partname, path, name,
-					 verity_options);
+				ret = asprintf(&command, "%s mount %s %s %s %s",
+					       script, partname, path, name,
+					       verity_options);
 
-			asprintf(&umount_cmd, "%s umount %s %s %s", script,
-				 partname, path, name);
+			ret = asprintf(&umount_cmd, "%s umount %s %s %s",
+				       script, partname, path, name);
+
+			if (ret < 0) {
+				pv_log(ERROR,
+				       " cannot allocate memory for mount/umount command");
+				return ret;
```

Comment:
At this point `handler` could have been allocated, so maybe it's better to "goto out" instead exit at this point using `return`